tests/key_field/key_field_sample_generated.h

```diff
@@ -170,13 +200,362 @@ inline bool operator!=(const Bar &lhs, const Bar &rhs) {
+        padding0__(0),
+        padding1__(0) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Color(flatbuffers::span<const float, 3> _rgb, uint8_t _tag)
+      : tag_(flatbuffers::EndianScalar(_tag)),
+        padding0__(0),
+        padding1__(0) {
+    flatbuffers::CastToArray(rgb_).CopyFromSpan(_rgb);
+    (void)padding0__;
+    (void)padding1__;
+  }
+  const flatbuffers::Array<float, 3> *rgb() const {
+    return &flatbuffers::CastToArray(rgb_);
+  }
+  flatbuffers::Array<float, 3> *mutable_rgb() {
+    return &flatbuffers::CastToArray(rgb_);
+  }
+  bool KeyCompareLessThan(const Color * const o) const {
+    return KeyCompareWithValue(o->rgb()) < 0;
+  }
+  int KeyCompareWithValue(const flatbuffers::Array<float, 3> *_rgb) const {
+    const flatbuffers::Array<float, 3> *curr_rgb = rgb();
+    for (flatbuffers::uoffset_t i = 0; i < curr_rgb->size(); i++) {
+      const auto lhs = curr_rgb->Get(i);
+      const auto rhs = _rgb->Get(i);
+      if(lhs != rhs)
+        return static_cast<int>(lhs > rhs) - static_cast<int>(lhs < rhs);
+    }
+    return 0;
+  }
+  uint8_t tag() const {
+    return flatbuffers::EndianScalar(tag_);
+  }
+  void mutate_tag(uint8_t _tag) {
+    flatbuffers::WriteScalar(&tag_, _tag);
+  }
+};
+FLATBUFFERS_STRUCT_END(Color, 16);
+
+inline bool operator==(const Color &lhs, const Color &rhs) {
+  return
+      (lhs.rgb() == rhs.rgb()) &&
+      (lhs.tag() == rhs.tag());
+}
+
+inline bool operator!=(const Color &lhs, const Color &rhs) {
+    return !(lhs == rhs);
+}
+
+
+FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Apple FLATBUFFERS_FINAL_CLASS {
+ private:
+  uint8_t tag_;
+  int8_t padding0__;  int16_t padding1__;
+  keyfield::sample::Color color_;
+
+ public:
+  static const flatbuffers::TypeTable *MiniReflectTypeTable() {
+    return AppleTypeTable();
+  }
+  Apple()
+      : tag_(0),
+        padding0__(0),
+        padding1__(0),
+        color_() {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  Apple(uint8_t _tag, const keyfield::sample::Color &_color)
+      : tag_(flatbuffers::EndianScalar(_tag)),
+        padding0__(0),
+        padding1__(0),
+        color_(_color) {
+    (void)padding0__;
+    (void)padding1__;
+  }
+  uint8_t tag() const {
+    return flatbuffers::EndianScalar(tag_);
+  }
+  void mutate_tag(uint8_t _tag) {
+    flatbuffers::WriteScalar(&tag_, _tag);
+  }
+  const keyfield::sample::Color &color() const {
+    return color_;
+  }
+  keyfield::sample::Color &mutable_color() {
+    return color_;
+  }
+  bool KeyCompareLessThan(const Apple * const o) const {
+    return KeyCompareWithValue(o->color()) < 0;
+  }
+  int KeyCompareWithValue(const keyfield::sample::Color  &_color) const {
+    const auto lhs_color = color();
+    const auto rhs_color = _color;
+    const auto rhs_color_rgb = rhs_color.rgb();
+    const auto rgb_compare_result = lhs_color.KeyCompareWithValue(rhs_color_rgb);
+    if (rgb_compare_result != 0)
+      return rgb_compare_result;
```

Comment:
In `KeyCompareWithValue ` of  **Apple** we compare both color and tag. (I believe compare of tag is unnecessary as color is key of Apple.)
But in  `KeyCompareWithValue ` of  **Color** we just compare rgb (color key) and ignore tag.
Which one is correct ?

Reply:
In `KeyCompareWithValue` of Color, I am comparing rgb first, and if rgb of lhs and rhs are the same, then comparing tag, I didn't ignore tag.